### PR TITLE
'migrate to launchTemplate'

### PR DESCRIPTION
**Pull request by calvix**

### CODE REVIEW

**Comment by MarcelMue** [comment]
It's pretty hard to tell what the actual change was with the renaming as well.

File: service/controller/resource/tcnp/template/template_main_auto_scaling_group.go [hunk 1]
```diff
@@ -16,7 +16,14 @@ const TemplateMainAutoScalingGroup = `
...
       MaxSize: {{ .AutoScalingGroup.MaxSize }}
-      LaunchConfigurationName: !Ref NodePoolLaunchConfiguration
+      MixedInstancesPolicy: 
+        LaunchTemplate:
+          LaunchTemplateSpecification:
+            LaunchTemplateName: !Ref NodePoolLaunchTemplate
+            Version:
+              Fn::GetAtt
+              - NodePoolLaunchTemplate
+              - LatestVersionNumber
```
Comment (MarcelMue) [inline service/controller/resource/tcnp/template/template_main_auto_scaling_group.go]: Why is this `LatestVersionNumber` needed and what does it do?

Comment (calvix) [reply service/controller/resource/tcnp/template/template_main_auto_scaling_group.go]: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-autoscaling-autoscalinggroup-launchtemplatespecification.html

**Comment by calvix** [comment]
the actual change is migrating to launch template

that is a new structure

File: service/controller/resource/tcnp/testdata/case-0-basic-test.golden [hunk 2]
```diff
@@ -22,7 +22,11 @@ Resources:
       DesiredCapacity: 3
```
Comment (calvix) [inline service/controller/resource/tcnp/testdata/case-0-basic-test.golden]: you can check the golden file diff to see the real changes

**Pull request merged**